### DISCUSSION TITLE
insert langserve into an existing fastapi?

### DISCUSSION THREAD

**Discussion by markat1** [langchain-ai/langserve#728] [Q&A] [answered]
So I can create a new app via [langchain-cli](https://github.com/langchain-ai/langchain/blob/master/templates/README.md ) for making langservice integrations with fastapi.

Though I want to implement langserve into an existing fastapi aplication.

**Question**: Is there a proper way to implement langserve into a existing fastapi without having to create a new app with langchain-cli?

**Comment by eyurtsev** [comment, marked as answer]
Yes, you can just use `add_routes` -- take a look at the examples associated with this repo
